ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -385,7 +385,13 @@ public static AttestationProcessingResult on_attestation(
     Attestation attestation = validateableAttestation.getAttestation();
 
     return validateOnAttestation(store, attestation, forkChoiceStrategy)
-        .ifSuccessful(() -> indexAndValidateAttestation(validateableAttestation, maybeTargetState))
+        .ifSuccessful(
+            () -> {
+              if (validateableAttestation.isValidIndexedAttestation()) {
```

Comment:
Instead of short-circuiting here, I think you can just use the new `is_valid_indexed_attestation` method in `indexAndValidateAttestation`

Reply:
Done, but I again had to check for target state being empty to return unknown block. So I again had to short circuit.